tests/test_preprocess_pipeline.py

```diff
@@ -0,0 +1,87 @@
+import os.path
+from unittest import TestCase
+
+import numpy as np
+
+from aspire.estimation.noise import WhiteNoiseEstimator
+from aspire.source.simulation import Simulation
+from aspire.utils.coor_trans import grid_2d
+from aspire.utils.filters import RadialCTFFilter, ScalarFilter
+from aspire.utils.matrix import anorm
+from aspire.volume import Volume
+
+DATA_DIR = os.path.join(os.path.dirname(__file__), 'saved_test_data')
+
+
+class PreprocessPLTestCase(TestCase):
+    def setUp(self):
+
+        self.L = 64
+        self.n = 128
+
+        self.noise_variance = 0.004
+        self.noise_filter = ScalarFilter(dim=2, value=self.noise_variance)
+
+        self.sim = Simulation(
+            L=self.L,
+            n=self.n,
+            unique_filters=[RadialCTFFilter(defocus=d) for d in np.linspace(1.5e4, 2.5e4, 7)],
+            noise_filter=self.noise_filter
+        )
+        self.imgs_org = self.sim.images(start=0, num=self.n).asnumpy()
```

Comment:
If you're getting all images, just use `self.sim.images()` instead of `self.sim.images(start=0, num=self.n)` throughout this code.